R/compute_mid_year_age.R

```diff
@@ -0,0 +1,18 @@
+#' Compute Age of Clients at Midpoint of Year
+#'
+#' @param data data to assign age to
+#' @param fyyear financial year
+#' @param dob date of birth of the client
+#'
+#' @return a vector of age at midpoint
+#' @export
+#'
+#' @examples
+#' dob <- as.Date(c("01-01-1990", "10-05-1960"))
+#' fyyear <- "1920"
+#' compute_mid_year_age(fyyear, dob)
+compute_mid_year_age <- function(data, fyyear, dob) {
+  data <- data %>%
+    # age
+    dplyr::mutate(age = lubridate::as.period(lubridate::interval(start = dob, end = midpoint_fy(fyyear)))$year)
+}
```

Comment:
Sorry, I think I've already checked this once, but could we just take an age and return a dob (as in the example) rather than taking and returning a data frame?
```suggestion
#' Compute Age at Midpoint of Year
#'
#' @param fyyear financial year
#' @param dob date of birth
#'
#' @return a vector of ages at the financial year midpoint
#' @export
#'
#' @examples
#' dob <- as.Date(c("01-01-1990", "10-05-1960"))
#' fyyear <- "1920"
#' compute_mid_year_age(fyyear, dob)
compute_mid_year_age <- function(fyyear, dob) {

 age_intervals <- lubridate::interval(start = dob, end = midpoint_fy(fyyear))

 ages <- lubridate::as.period(age_intervals$year)
 
 return(ages)
}
```